hive-metastore/src/main/java/org/apache/iceberg/hive/CachedClientPool.java

```diff
@@ -87,4 +91,23 @@ public <R> R run(Action<R, IMetaStoreClient, TException> action, boolean retry)
       throws TException, InterruptedException {
     return clientPool().run(action, retry);
   }
+
+  @VisibleForTesting
+  @Value.Immutable
+  abstract static class Key {
```

Comment:
This is great, this is actually a common issue, a lot of values are not in the cache key but should be..

See this issue as well:  https://github.com/apache/iceberg/pull/5378  for default Hive catalog, I guess we can put catalog in this as well then.

Reply:
Yeah, our use case here only needs the UGI, we can extend the cache key to take more values as needed.